### ISSUE TITLE
bson_malloc memory leak

### ISSUE THREAD

**Issue by valeriomazzeo** [mongodb/mongo-swift-driver#458]
It seems that some Documents are not released from within the `CreateIndexesOperation`.

![Screenshot 2020-04-27 at 15 17 57](https://user-images.githubusercontent.com/1882080/80382922-d3531500-889a-11ea-95d2-03229b7983a5.png)

### Swift version

What does this command give you?
```
$ swift --version
Apple Swift version 5.1.3 (swiftlang-1100.0.282.1 clang-1100.0.33.15)
Target: x86_64-apple-darwin19.4.0
```
### Operating system

What does this command give you?
```
$ uname -a
Darwin MBP 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```
### Driver version

What does this command give you?
```
$ cat Package.resolved # Applies if you are using Swift package manager
{
        "package": "bson",
        "repositoryURL": "https://github.com/mongodb/swift-bson",
        "state": {
          "branch": null,
          "revision": "40f770ed265bb9958edbcd0c809a3c2afe75a3ba",
          "version": "2.1.0"
        }
},
{
        "package": "MongoSwift",
        "repositoryURL": "https://github.com/mongodb/mongo-swift-driver.git",
        "state": {
          "branch": null,
          "revision": "1b38ca8987d1938e0c9245f959fa2172b13cba60",
          "version": "0.2.0"
        }
}
```

### What is the version(s) of `mongod` that you are running with the driver?
Try running:
```
$ mongod --version
4.2.2
```

**Comment by kmahar** [comment]
Hi @valeriomazzeo, thanks for the report!

Could you provide any more info on the code that triggered this, or a code sample? We have a number of index creation methods and it would be useful to know e.g. which is being used here, which options were specified for the command if any, etc.

Also, I don't develop in Xcode much so I'm not quite sure how to get to the view / tool you've shown in the screenshot, any more details on that would be very helpful.

Thanks!

**Comment by valeriomazzeo** [comment]
The trigger from my driver should be [this one](https://github.com/asensei/vapor-fluent-mongo/blob/master/Sources/FluentMongo/MongoConnection.swift#L177), which leads to `MongoCollection+Indexes.swift` / `createIndexes` with model.

```swift
    @discardableResult
    public func createIndex(
        _ model: IndexModel,
        options: CreateIndexOptions? = nil,
        session: ClientSession? = nil
    ) throws -> String {
        return try self.createIndexes([model], options: options, session: session)[0]
    }
```

`IndexOptions` could be `unique` or just `nil`/`empty`.

That screenshot is coming from Instruments, with the Leaks template.
You can build your project for Profiling, then record the execution using Instruments, and it will detect Memory Leaks showing you that stack trace. You'd have to somehow trigger that piece of code at runtime in order for instruments to profile it.

Please let me know if you need any more info. I think it's just the case of a `Document` missing a release somewhere.

**Comment by kmahar** [comment]
Hey @valeriomazzeo, just wanted to provide a status update on this: thanks for the details, I've been able to reproduce this. We're passing the `bson_t` to `bson_destroy` correctly, so I'm working with the C driver team now to figure out what's going on in there.
I'll let you know what I find out.

**Comment by eriadam** [comment]
I am seeing this also with write operations.
<img width="406" alt="Screenshot 2020-05-04 at 16 30 52" src="https://user-images.githubusercontent.com/2480660/80977260-c4cda600-8e24-11ea-8a32-f601d29c988a.png">

**Comment by patrickfreed** [comment]
Quick update: we've identified the issue and filed [SWIFT-824](https://jira.mongodb.org/browse/SWIFT-824) to track the work for fixing it. The fix should be included in the next 1.0 release candidate.

**Comment by kmahar** [comment]
This has been fixed by #474 so I'm going to go ahead and close the issue. Thanks for the reports @valeriomazzeo @eriadam !
